Namespace 'global.Express' has no exported member 'SessionData' with TypeScript build

Note: In addition to stopping my own project from building w/o an override, this seems to be the nature of the problem Yoshi is flagging at #245.

By inspection, it appears the `SessionData` interface was moved from `Express` to a more narrowly scoped namespace in the `express-session` middleware, quite some time ago. I THINK this is therefore a relatively trivial fix (change `Express.SessionData` to `expressSession.SessionData` on these lines):

https://github.com/googleapis/nodejs-datastore-session/blob/7f0a56193bb1aead7f4448887064e0e4d04033a3/src/index.ts#L58 

and 

https://github.com/googleapis/nodejs-datastore-session/blob/7f0a56193bb1aead7f4448887064e0e4d04033a3/src/index.ts#L99

After a bit of code anthropology, however, I'm wondering about backward compatibility. In particular, what's not yet clear to me is why this is (apparently) a new issue, although it is clearly a recently prolific one (not unique to this module - see also https://stackoverflow.com/questions/64845125/namespace-express-has-no-exported-member-sessiondata), and I also anticipate this commit to be very relevant:

https://github.com/DefinitelyTyped/DefinitelyTyped/commit/d1259ee31e6b17c31a5a86e27a0d12f3ec7c5a19#comments

Despite the DefintelyTyped changes above, I never found a definitive point around that time at which the `SessionData` interface was removed from the global `Express` namespace. If I can figure that out, and by chance I still have a valid contributor agreement w/Google, and I can quickly figure out how to run the build and tests, and no one else beats me to it, I'll dig in and hopefully then submit a PR to follow, but that's sure a lot of 'ifs' so no promises.


#### Environment details

  - OS: local: macOS Big Sur 11.0.1
  - Node.js version: v12.13.1
  - npm version: 6.14.5
  - TypeScript tsc version: 4.0.5
  - `@google-cloud/connect-datastore` version: 5.0.1

#### Steps to reproduce

As this appears to be a TypeScript compile issue, so most of these steps are setup.
Generate a minimal TypeScript Express app, using this module for the express connect session store.

  1. `npm init` (then use an arbitrary package name, version, description, entry point, etc.)
  2. `npm install --save express @types/express express-session @types/express-session @google-cloud/datastore @google-cloud/connect-datastore` 
  3. Make a couple of common subdirectories - `mkdir src dist`
  4. Save the attached `tsconfig.json` in the root.
  5. Save the attached `index.ts` file under `src`.
  6. Run the typescript compiler using `tsc -p .`

Actual output:

node_modules/@google-cloud/connect-datastore/build/src/index.d.ts:14:72 - error TS2694: Namespace 'global.Express' has no exported member 'SessionData'.

14     get: (sid: string, callback: (err: Error | null, session?: Express.SessionData | null | undefined) => void) => void;
                                                                          ~~~~~~~~~~~

node_modules/@google-cloud/connect-datastore/build/src/index.d.ts:15:38 - error TS2694: Namespace 'global.Express' has no exported member 'SessionData'.

15     set: (sid: string, sess: Express.SessionData, callback?: ((err: Error | null) => void) | undefined) => void;
                                        ~~~~~~~~~~~


Found 2 errors.



tsconfig.json:
```
{
  "compilerOptions": {
      "module": "commonjs",
      "lib": ["ES2018"],
      "esModuleInterop": true,
      "allowSyntheticDefaultImports": true,
      "target": "es6",
      "noImplicitAny": true,
      "moduleResolution": "node",
      "sourceMap": true,
      "outDir": "dist",
      "baseUrl": ".",
      "paths": {
          "*": [
              "node_modules/*",
              "src/types/*"
          ]
      }
  },
  "include": [
    "src/**/*"
  ],
  "exclude": [
    "src/public"
  ]
}
```

src/index.ts:
```
'use strict';

import { default as express, Request, Response, NextFunction } from "express";
import { Datastore } from "@google-cloud/datastore";
import { DatastoreStore } from "@google-cloud/connect-datastore";
import { default as session } from "express-session";

declare var process : {
  env: {
    PORT: number,
  }
}

const app = express();
const port = process.env.PORT || 8080;

const datastore = new Datastore();

const datastorestore = new DatastoreStore({
  dataset: datastore
});

const sessionMiddleware = session({
  cookie: {
    httpOnly: true,
    secure: true,
  },
  name: 'not-a-real-session-cookie-name',
  resave: false,
  saveUninitialized: false,
  store: datastorestore,
  secret: 'not-a-real-session-secret'
});

app.set('trust proxy', 1); // trust first proxy (required because app engine terminates TLS before us)

app.use(sessionMiddleware);

app.get('/', async (req: Request, res: Response, next: NextFunction) => {
  try {
    res.sendStatus(200);
  } catch(error) {
    next(error);
  }
});

// Start the server
app.listen(port, () => {
  console.log(`App listening on port ${port}`);
  console.log('Press Ctrl+C to quit.');
});
```

--- a/src/index.ts
+++ b/src/index.ts
@@ -31,7 +31,7 @@ export class DatastoreStore extends expressSession.Store {
   kind: string;
   expirationMs: number;
   constructor(options: DataStoreOptions = {}) {
-    super(options);
+    super();
     this.ds = (options.dataset || options.datastore)!;
     if (!this.ds) {
       throw new Error('No dataset provided to Datastore Session.');
@@ -55,7 +55,10 @@ export class DatastoreStore extends expressSession.Store {
 
   get = (
     sid: string,
-    callback: (err: Error | null, session?: Express.SessionData | null) => void
+    callback: (
+      err: Error | null,
+      session?: expressSession.SessionData | null
+    ) => void
   ) => {
     this.ds.get(this.ds.key([this.kind, sid]), (err, entity) => {
       if (err) {
@@ -96,7 +99,7 @@ export class DatastoreStore extends expressSession.Store {
 
   set = (
     sid: string,
-    sess: Express.SessionData,
+    sess: expressSession.SessionData,
     callback?: (err: Error | null) => void
   ) => {
     callback = callback || (() => {});

--- a/system-test/session.ts
+++ b/system-test/session.ts
@@ -16,6 +16,7 @@ import * as assert from 'assert';
 import {describe, it} from 'mocha';
 import {DatastoreStore} from '../src';
 import {Datastore} from '@google-cloud/datastore';
+import {SessionData} from 'express-session';
 
 describe('works with no expiration', () => {
   const store = new DatastoreStore({
@@ -31,7 +32,7 @@ describe('works with no expiration', () => {
   });
 
   it('Should create and retrieve a session', done => {
-    store.set('id1', ({foo: 'bar'} as {}) as Express.SessionData, err => {
+    store.set('id1', ({foo: 'bar'} as {}) as SessionData, err => {
       assert.ifError(err);
       store.get('id1', (err, session) => {
         assert.ifError(err);
@@ -61,7 +62,7 @@ describe('expired session is not returned', () => {
   });
 
   it('Should create but not retrieve an expired session', done => {
-    store.set('id2', ({foo: 'bar'} as {}) as Express.SessionData, err => {
+    store.set('id2', ({foo: 'bar'} as {}) as SessionData, err => {
       assert.ifError(err);
       store.get('id2', (err, session) => {
         assert.ifError(err);
@@ -79,7 +80,7 @@ describe('unexpired session is returned', () => {
   });
 
   it('Should create and retrieve a session', done => {
-    store.set('id3', ({foo: 'bar'} as {}) as Express.SessionData, err => {
+    store.set('id3', ({foo: 'bar'} as {}) as SessionData, err => {
       assert.ifError(err);
       store.get('id3', (err, session) => {
         assert.ifError(err);

